### ISSUE TITLE
Germany - only Apple WiFi backend works

### ISSUE THREAD

**Issue by herrdeh** [microg/android_packages_apps_UnifiedNlp#33]
Hi everybody,
great approach - I really appreciate it!

Installed ug UnifiedNlp via f-droid on a Huawei y300, slimkat 4.4.4. No trouble - and with the Apple WiFi backend I get instant location data.

But:
I couldnt make any other backend work. Are there insufficient data for Berlin, Germany? - Or how can I troubleshoot the backends? - No idea where I could find error messages.

On the "Here GPS location" app I get "network not enabled", allthough location services are switched on and are set to "energy saving mode".

Btw: 
Wouldnt it a good idea to split GSM and WiFi databases by country / continent - as map providers do? - And combine it with an option to load missing data online? - 400MB is not so little - and it's quite likely I'll not travel to US in the next few years...

Cheers,
Wolf

**Comment by n76** [comment]
Which other backends did you try?

With respect to the two backends that I wrote:
https://github.com/n76/Local-GSM-Backend has a field for setting the mobile country code (MCC) value(s) to store when downloading a new database. It defaults to the United States (301,311) but if you set it to 262 it should save the towers for Germany. You can specify multiple codes for instance use 262,208 for Germany and France.

For https://github.com/n76/wifi_backend the data is collected by your phone when the GPS is in use by some other app. "Out of the box" it has no data at all and cannot give you an estimated location. I usually run a mapping or navigation application (OSMtracker, OsmAnd, etc.) when I am going somewhere so when I am ready to leave the destination the backend will have learned the WiFi APs in that area and I get a fast location estimate and quick GPS lock.

**Comment by herrdeh** [comment]
Thanks for the hint on '262' - I'll try that first.

At the moment I'm trying to get the non-gapps version, but can't install 
it. F-droid ever states that a prior version has to be removed first. But I 
deleted anything I had installed before.

So what can I do?

Gesendet mit AquaMail für Android
http://www.aqua-mail.com


Am 6. März 2015 16:46:17 schrieb n76 <notifications@github.com>:

**Comment by n76** [comment]
Not sure about your issue as I've only ever had the non-gapps version on my phone.

I neglected to give a link to a list of mobile country codes in my previous response. Here is an English version: http://en.wikipedia.org/wiki/Mobile_country_code

**Comment by herrdeh** [comment]
I found, that I have to remove the /data/data/com.google.android.gms-1.apk directory in order to make it work. But it looks as if the non-gapps UnifiedNlp doesnt cohabitate with frozen (by Titanium Backup) play services and disappears from the app drawer.

So I installed the with-gapps version again - but that doesnt work without *running* google play services, and my idea of salvaging a-bit-of-google on my phone wont work. 

But does it make sense to use ug UnifiedNlp at all, if you have to maintain google play services anyway?

**Comment by mar-v-in** [comment]
> But does it make sense to use ug UnifiedNlp at all, if you have to maintain google play services anyway?

You might still use a backend that runs offline and are not forced to always send details of your location to Google.

**Comment by herrdeh** [comment]
Good point! But how do I ensure that Google is not calling home?

Gesendet mit AquaMail für Android
http://www.aqua-mail.com

**Comment by mar-v-in** [comment]
Only way to completely ensure this, is to wipe off all GAPPS from the phone (= wipe /system from recovery, install custom rom, don't install any GAPPS packages, reboot and uninstall any Google Apps still installed).

If you just want less: using UnifiedNlp with Xposed module will cause the system to no longer use google's location provider, disabling all location options in Google settings should stop it from sending out your location afterwards.

**Comment by herrdeh** [comment]
I ended up with completely erasing google (well - at my phone only...).
That made my cheap Huawei so much more responsive and agile, and so much improved battery duration (once had round 40 hrs at quite frequent usage), that it's a very good trade.

So I tried your suggestion and "calibrated" a trip with GPS active. But I found that GPS will deactivate automatically if not e.g. a navigator app is calling it. After a time I saw at GPS monitor that GPS is again seeking for position data "from scratch". - So - is there a possibility to keep GPS alive?

Anyway: Thanks a lot for providing UnifiedNLP!!

**Comment by mar-v-in** [comment]
Using [Mozilla Stumbler](https://f-droid.org/repository/browse/?fdfilter=mozilla&fdid=org.mozilla.mozstumbler) will keep GPS on and collect data that can be used to submit to the Mozilla Location Service (that can be used as a backend as well)

**Comment by KAMiKAZOW** [comment]
If you want to support a non-proprietary service, use http://www.radiocells.org/ and http://www.openwlanmap.org/
Both have NLP back-ends, although the latter's is currently not in F-Droid.

**Comment by herrdeh** [comment]
AFAIK, there is quite a number of open/free location services. Do they merge/share their data bases regularly? - And how do I know which one to support for best efficiency?

**Comment by mar-v-in** [comment]
To my knowledge, Mozilla Location Services and OpenCellId synchronize their cell databases.

Mozilla's cell database is under CC0 license and OpenCellId is CC-BY-SA licensed. Mozilla's Wi-Fi database is not published for legal reasons: In some countries a combination of location, mac address and timestamp is considered private data and thus cannot be released.

**Comment by IzzySoft** [comment]
> If you just want less: using UnifiedNlp with Xposed module will cause the system to no longer use google's location provider, disabling all location options in Google settings should stop it from sending out your location afterwards.

I've tried that: Installed XPosed, installed the module, activated it, rebooted. Then tried to configure it from within Xposed (click image for larger variant):

[![xposed](https://cloud.githubusercontent.com/assets/6781438/7901624/6b4c9aba-0791-11e5-81a4-3395785604c3.jpg)](https://cloud.githubusercontent.com/assets/6781438/7901603/7d28680a-0790-11e5-973c-abe256e0c9e9.png)  

So it looks it doesn't work? On that device (LG P880) I still have the stock ROM running (4.1.2), so all the GApps are there. Installed UnifiedNLP with the LocalGSMBackend, and have a CellID database in place (as the configuration page of LocalGSMBackend confirms). I cannot install LegacyNetworkLocation.apk as its package name (`com.google.android.location`) would collide with an already installed package. Any ideas?

> To my knowledge, Mozilla Location Services and OpenCellId synchronize their cell databases.

I don't think so. I'm using [lacells-creator](https://github.com/n76/lacells-creator) to create the database. It has a script to merge from both sources. The downloaded CSVs differ in size by ~20% (btw: that very same database works fine on my other, Google-free device – so I can rule it out for above issue).

**Comment by IzzySoft** [comment]
@mar-v-in Any idea on my previous comment? I still cannot get it to work. That "make sure you added at least one backend" message I saw only once; usually it's rather the first screenshot I get to see. I've checked all other issues here to make sure I didn't miss anything:

* #22 I've explicitly entered the app before and verified the database is recognized – so it was definitely started
* #34 I've checked for details there (and definitely agree: could not find a step-by-step description for devices with (stock ROMs and) GApps installed – pointers welcome)
* #9 didn't help me either. As the OP there, I still need GCM notifications, so uninstalling GApps is no way here. In *Settings › Apps* I can see *µg UnifiedNlp* and *XposedUnifiedNlp*, *GSM Location Service* shows up there as well. The latter also shows up in the first, and is enabled (see screenshot above).
* very same cell database is working fine on my other device (CyanogenMod with NOGAPPS), so that cannot be it

I'm out of ideas. Will be abroad for a few days next, with "no data" while on the road (i.e. no mobile data, though WiFi is available at the hotel). As that means Google network location itself will have no chance to work, but the local database should, I will surely cross-check there. Could well be the Google stuff is just overriding things here – though according to the docs, the Xposed Module should take care for that itself. But as even initialization checks fail there, I don't get any preference screen I could cross-check with.

If you want me to create a new issue for this, please let me know. As said, I'm abroad for the next 10 days (was hoping to have this fixed before), so I could do so on my return.
